pallets/liquidity-proxy/src/lib.rs

```diff
@@ -1545,6 +1549,185 @@ impl<T: Config> Pallet<T> {
+        fallible_iterator::convert(receivers.into_iter().map(|val| Ok(val))).for_each(|receiver| {
+            // extrinsic fails if there are duplicate account ids in the same output asset id
+            if !unique_batch_receivers.insert(receiver.account_id.clone()) {
+                fail!(Error::<T>::AggregationError);
+            }
+            assets::Pallet::<T>::transfer_from(
+                &output_asset_id,
+                &sender,
+                &receiver.account_id,
+                receiver.target_amount - remainder_per_receiver,
+            )
+        })
+    }
+
+    fn calculate_adar_commission(amount: Balance) -> Result<Balance, DispatchError> {
+        let adar_commission_ratio = FixedWrapper::from(ADAR_COMMISSION_RATIO);
+
+        let adar_commission = (FixedWrapper::from(amount) * adar_commission_ratio)
+            .try_into_balance()
+            .map_err(|_| Error::<T>::CalculationError)?;
+
+        Ok(adar_commission)
+    }
+
+    fn inner_swap_batch_transfer(
+        sender: &T::AccountId,
+        input_asset_id: &T::AssetId,
+        swap_batches: Vec<SwapBatchInfo<T::AssetId, T::DEXId, T::AccountId>>,
+        mut max_input_amount: Balance,
+        selected_source_types: &Vec<LiquiditySourceType>,
+        filter_mode: &FilterMode,
+    ) -> Result<Balance, DispatchError> {
+        let mut unique_asset_ids: BTreeSet<T::AssetId> = BTreeSet::new();
+
+        let mut adar_commission = balance!(0);
+
+        fallible_iterator::convert(swap_batches.into_iter().map(|val| Ok(val))).for_each(
+            |swap_batch_info| {
+                let SwapBatchInfo {
+                    outcome_asset_id: asset_id,
+                    dex_id,
+                    receivers,
+                } = swap_batch_info;
+
+                // extrinsic fails if there are duplicate output asset ids
+                if !unique_asset_ids.insert(asset_id.clone()) {
+                    fail!(Error::<T>::AggregationError);
+                }
+
+                if receivers.len() == 0 {
+                    fail!(Error::<T>::InvalidReceiversInfo);
+                }
+
+                let out_amount = receivers.iter().map(|recv| recv.target_amount).sum();
+
+                let (mut executed_input_amount, remainder_per_receiver): (Balance, Balance) =
+                    if &asset_id != input_asset_id {
+                        Self::exchange_batch_tokens(
+                            &sender,
+                            receivers.len() as u128,
+                            &input_asset_id,
+                            &asset_id,
+                            max_input_amount,
+                            &selected_source_types,
+                            dex_id,
+                            &filter_mode,
+                            out_amount,
+                        )?
+                    } else {
+                        (out_amount, 0)
+                    };
+
+                let adar_commssion_portion =
```

Comment:
Why not calculate sum of input amounts and call `calculate_adar_commission` on this number?